src/functional/type_system/type_info.py

```diff
@@ -227,22 +227,22 @@ def is_arithmetic(symbol_type: ts.TypeSpec) -> bool:
     return is_floating_point(symbol_type) or is_integral(symbol_type)
 
 
-def is_field_type_or_tuple_of_field_type(type_: ts.TypeSpec) -> bool:
+def is_type_or_tuple_of_type(type_: ts.TypeSpec, expected_type: type | tuple) -> bool:
     """
-     Return True if ``type_`` is FieldType or FieldType nested in TupleType.
+    Return True if ``type_`` matches any of the expected.
 
-     Examples:
-     ---------
+    Examples:
+    ---------
     >>> scalar_type = ts.ScalarType(kind=ts.ScalarKind.INT)
     >>> field_type = ts.FieldType(dims=[], dtype=scalar_type)
-     >>> is_field_type_or_tuple_of_field_type(field_type)
-     True
-     >>> is_field_type_or_tuple_of_field_type(ts.TupleType(types=[field_type, field_type]))
-     True
-     >>> is_field_type_or_tuple_of_field_type(ts.TupleType(types=[field_type, scalar_type]))
-     False
+    >>> is_type_or_tuple_of_type(field_type, ts.FieldType)
+    True
+    >>> is_type_or_tuple_of_type(ts.TupleType(types=[scalar_type, field_type]), (ts.ScalarType, ts.FieldType))
+    True
+    >>> is_type_or_tuple_of_type(scalar_type, ts.FieldType)
+    False
```

Comment:
```suggestion
    False
    >>> is_type_or_tuple_of_type(ts.TupleType(types=[scalar_type, field_type]), ts.FieldType)
    False
```
just for completness